Makefile

```diff
@@ -8,35 +8,26 @@ mypy:
 	mypy --ignore-missing-imports $(MODULES)
 
 test_srcs := $(wildcard tests/test_*.py)
-all_test_srcs := $(addprefix all__, $(test_srcs))
-integration_test_srcs := $(addprefix integration__, $(test_srcs))
 
-test: lint mypy $(test_srcs)
+define test_rules
+$1_test_srcs := $$(addprefix $1__, $$(test_srcs))
+$1_test: lint mypy $$($1_test_srcs)
 	coverage combine
 	rm -f .coverage.*
 
-$(test_srcs): %.py :
-	set -o pipefail \
-	&& DSS_TEST_MODE="standalone" coverage run -p --source=dss -m unittest $(DSS_UNITTEST_OPTS) $@  2>&1 \
-	| sed -e "s/^/$$$$ /"
+$$($1_test_srcs): $1__%.py :
+	@set -o pipefail \
+	&& (DSS_TEST_MODE="$2" coverage run -p --source=dss -m unittest $$(DSS_UNITTEST_OPTS) $$*.py) 2>&1 \
+	| sed -e "s/^/$$$$$$$$ /"
 
-integration_test: lint mypy $(integration_test_srcs)
-	coverage combine
-	rm -f .coverage.*
-
-$(integration_test_srcs): integration__%.py :
-	set -o pipefail \
-	&& DSS_TEST_MODE="integration" coverage run -p --source=dss -m unittest $(DSS_UNITTEST_OPTS) $*.py 2>&1 \
-	| sed -e "s/^/$$$$ /"
+.PHONY: $1_test $$($1_test_srcs)
+endef
 
-all_test: lint mypy $(all_test_srcs)
-	coverage combine
-	rm -f .coverage.*
+$(eval $(call test_rules,standalone,standalone))
+$(eval $(call test_rules,integration,integration))
+$(eval $(call test_rules,all,integration standalone))
```

Comment:
The tests aren't actually run serially anymore. That was in an earlier version of this PR and I backed *that* change out after @ttung pointed at an unintended side-effect. I left *this* change in because while increasing complexity, it reduces duplication. So in a way it is more magic, less alphabet soup.

Reply:
We need to refactor our Makefile to make it readable.

Reply:
No such thing as a readable Makefile. But I'll back this out. I think it's an improvement because it is less alphabet soup, but more delicious. Only half kidding.